controls/V-72315.rb

```diff
@@ -77,7 +147,48 @@
 specific hosts.   "
   tag "fix_id": "F-78669r2_fix"
 
-  describe "This control must be reviewed manually" do
-    skip "You must review this control manually."
+  if service('firewalld').running?
+    @default_zone = firewalld.default_zone
+
+    describe firewalld.where{ zone = @default_zone } do
+      its('services') { should be_in firewalld_services }
+    end
+
+    describe firewalld do
+      firewalld_hosts_allow.each do |rule|
+        it { should have_rule_enabled(rule) }
+      end
+      firewalld_hosts_deny.each do |rule|
+        it { should_not have_rule_enabled(rule) }
+      end
+      firewalld_ports_allow.each do |port|
+        it { should have_port_enabled_in_zone(port) }
+      end
+      firewalld_ports_deny.each do |port|
+        it { should_not have_port_enabled_in_zone(port) }
+      end
+    end
+  elsif service('iptables').running?
+    describe iptables do
+      iptable_rules.each do |rule|
+        it { should have_rule(rule) }
+      end
+    end
+  else
+    describe package('tcp_wrappers') do
+      it { should be_installed }
+    end
+    tcpwrappers_allow.each do |rule|
+      describe etc_hosts_allow.where { daemon == rule['daemon'] } do
+        its('client_list') { should be rule['client_list'] }
+        its('options') { should be rule['options'] }
+      end
+    end
+    tcpwrappers_deny.each do |rule|
```

Comment:
`/etc/hosts.deny` shouldn't exist at all should it?

Reply:
I think we need that check. The control's check-text says:

"If \"firewalld\" is not active, determine whether \"tcpwrappers\" is being used
by checking whether the \"hosts.allow\" and \"hosts.deny\" files are empty with
the following commands:"

Reply:
@djhaynes Ah, right...the default for `/etc/hosts.deny` should be `ALL: ALL`. Had to re-read the man page.

Reply:
Given this, I am going to merge in this control.